src/sentry/models/repository.py

```diff
@@ -93,6 +97,50 @@ def reset_pending_deletion_field_names(
         del self.config["pending_deletion_name"]
         return super().reset_pending_deletion_field_names(["config"])
 
+    def cascade_delete_on_hide(self) -> None:
+        # Manually cause a deletion cascade after setting a repo's status
+        # to ObjectStatus.HIDDEN.
+        # References RepositoryDeletionTask and BaseDeletionTask logic.
+
+        if self.status != ObjectStatus.HIDDEN:
+            return
+
+        from sentry.deletions import default_manager
+        from sentry.deletions.base import ModelRelation
+
+        def _delete_children(relations: list[ModelRelation]) -> bool:
+            for relation in relations:
+                task = default_manager.get(
+                    transaction_id=None,
+                    actor_id=None,
+                    task=relation.task,
+                    **relation.params,
+                )
+
+                has_more = True
+                while has_more:
+                    has_more = task.chunk()
+                    if has_more:
+                        metrics.incr("deletions.should_spawn", tags={"task": type(task).__name__})
+            return False
+
+        has_more = True
+
+        while has_more:
+            # get child relations
+            child_relations = [
+                ModelRelation(Commit, {"repository_id": self.id}),
+                ModelRelation(PullRequest, {"repository_id": self.id}),
+                ModelRelation(RepositoryProjectPathConfig, {"repository_id": self.id}),
```

Comment:
I think you will be better of figuring out how to share this logic. Otherwise someone is going to add a new relation to `deletions` at some point and miss adding this here.

Tbh I think you should figure out how to extract most of the logic that is copied here

Reply:
i can pull out the reused logic from the class into functions that both the `TaskManager` class and `Repository` model class call